### PR TITLE
Polish en doc of data_loader

### PR DESCRIPTION
**Pull request by zhiqiu**
image

### CODE REVIEW

File: doc/fluid/user_guides/howto/prepare_data/use_py_reader_en.rst [hunk 1]
```diff
@@ -1,285 +1,266 @@
...
-Use PyReader to read training and test data
+Asynchronous Data Reading
 ############################################
 
-Besides Python Reader, we provide PyReader. The performance of PyReader is better than :ref:`user_guide_use_numpy_array_as_train_data_en` , because the process of loading data is asynchronous with the process of training model when PyReader is in use. And PyReader can coordinate with :code:`double_buffer_reader` to improve the performance of reading data. What's more, :code:`double_buffer_reader` can achieve the transformation from CPU Tensor to GPU Tensor, which improve the efficiency of reading data to some extent.
+Besides Python Reader, we provide DataLoader. The performance of DataLoader is better than :ref:`user_guide_use_numpy_array_as_train_data_en` , because data reading and model training process is asynchronous
+when DataLoader is in use, and it can cooperate with :code:`double_buffer_reader` to improve the performance of reading data. What's more, :code:`double_buffer_reader` can achieve the asynchronous transformation from CPU Tensor to GPU Tensor, which improve the efficiency of reading data to some extent.
```
Comment (sneaxiy) [inline doc/fluid/user_guides/howto/prepare_data/use_py_reader_en.rst]: `which improve` -> `which improves`.

File: doc/fluid/user_guides/howto/prepare_data/use_py_reader_en.rst [hunk 2]
```diff
@@ -1,285 +1,266 @@
...
-Attention: If you want to create multiple PyReader objects（such as two different PyReader in training and inference period respectively）， you have to appoint different names for different PyReader objects,since PaddlePaddle uses different names to distinguish different variables, and `Program.clone()` (reference to :ref:`api_fluid_Program_clone` ）can't copy PyReader objects.
+    data_loader = fluid.io.DataLoader.from_generator(
+        feed_list=[image, label], capacity=64, use_double_buffer=True, iterable=ITERABLE)
 
-.. code-block:: python
+In the code, 
+
+- ``feed_list`` is the list of input variables; 
+- ``capacity`` is the buffer size of the DataLoader object in batches; 
+- ``use_double_buffer`` is True by default, which means ``double_buffer_reader`` is used. It is Recommended to use, because it can improve data reading speed;
```
Comment (sneaxiy) [inline doc/fluid/user_guides/howto/prepare_data/use_py_reader_en.rst]: `Recommended` -> `recommended`

File: doc/fluid/user_guides/howto/prepare_data/use_py_reader_en.rst [hunk 3]
```diff
@@ -1,285 +1,266 @@
...
-        paddle.reader.shuffle(paddle.batch(mnist.train(), 512), buf_size=8192))
+    # Configure data source of DataLoader.
+    places = fluid.cuda_places() if ITERABLE else None
+
+    train_loader.set_sample_list_generator(
+        fluid.io.shuffle(fluid.io.batch(mnist.train(), 512), buf_size=1024), places=places)
+
+    test_loader.set_sample_list_generator(fluid.io.batch(mnist.test(), 512), places=places)
+
+- Step 2, we choose different ways to run the network according to whether the DataLoader object is Iterable or not.
```
Comment (sneaxiy) [inline doc/fluid/user_guides/howto/prepare_data/use_py_reader_en.rst]: `Iterable` -> `iterable`

File: doc/fluid/user_guides/howto/prepare_data/use_py_reader_en.rst [hunk 4]
```diff
@@ -1,285 +1,266 @@
...
+
+    train_loader.set_sample_list_generator(
+        fluid.io.shuffle(fluid.io.batch(mnist.train(), 512), buf_size=1024), places=places)
+
+    test_loader.set_sample_list_generator(fluid.io.batch(mnist.test(), 512), places=places)
+
+- Step 2, we choose different ways to run the network according to whether the DataLoader object is Iterable or not. 
 
-    test_reader.decorate_paddle_reader(paddle.batch(mnist.test(), 512))
+If :code:`iterable = True`, the DataLoader object is a Python generator that can iterate directly useing for-range. The results returned by for-range are passed to the executor through the ``feed`` parameter of ``exe.run()``.
```
Comment (sneaxiy) [inline doc/fluid/user_guides/howto/prepare_data/use_py_reader_en.rst]: `useing` -> `using`

File: doc/fluid/user_guides/howto/prepare_data/use_py_reader_en.rst (hunk 1 above)
Comment (zhiqiu) [reply doc/fluid/user_guides/howto/prepare_data/use_py_reader_en.rst]: Done, thanks.

File: doc/fluid/user_guides/howto/prepare_data/use_py_reader_en.rst (hunk 2 above)
Comment (zhiqiu) [reply doc/fluid/user_guides/howto/prepare_data/use_py_reader_en.rst]: Done, thanks.

File: doc/fluid/user_guides/howto/prepare_data/use_py_reader_en.rst (hunk 3 above)
Comment (zhiqiu) [reply doc/fluid/user_guides/howto/prepare_data/use_py_reader_en.rst]: Done, thanks.

File: doc/fluid/user_guides/howto/prepare_data/use_py_reader_en.rst (hunk 4 above)
Comment (zhiqiu) [reply doc/fluid/user_guides/howto/prepare_data/use_py_reader_en.rst]: Done, thanks.

**Comment by zhiqiu** [comment]
follow comments.
image

**Review by sneaxiy — APPROVED** [review]

**Comment by zhiqiu** [comment]
change fluid.layers.data() to fluid.data()

image

**Review by lanxianghit — APPROVED** [review]

**Pull request merged**